src/lib/api-client.js

```diff
@@ -115,3 +115,11 @@ export function searchStudent(id){
         Dispatcher.handleAction('NO_STUDENT_FOUND');
     });
 }
+
+export function getAllModels(){
+    get('model/all', {}
+    ).then(data => {
+        Dispatcher.handleAction('MODELS_RECEIVED', data);
+        hashHistory.push('/models');
```

Comment:
If it's fine here, then we're all good.  It seems to me like all the consequences of an action should be handled within the action itself, but I get what you mean about scope.